### PR TITLE
feat: initialise a new totals tab with basic UI

### LINKED ISSUE #93: Feature request: group spending tab similar to splitwise
I have written my own script to calculate this from the json dump that spliit generates. But would be very useful if the same is shown on the UI itself
ref:

### PR DESCRIPTION
**Pull request by meta-boy**
A new tab named 'Stats' with a simple UI that shows the group-level spending summary. Can be further extended to show other stats, charts, etc.
closes #93 

Preview:

image

### CODE REVIEW

**Comment by justcallmelarry** [comment]
I don't have much to contribute here, but I thought I would suggest using "Stats" (or similar) instead of "Totals" as you could then expand it with other things, like spending charts etc raised in https://github.com/spliit-app/spliit/issues/82

**Comment by meta-boy** [comment]
Oh I agree. That would be much better. I was just making it closer to what splitwise has to offer but definitely could go with "Stats" and further add spending charts to it like you mentioned

**Comment by justcallmelarry** [comment]
I'm also wondering if this takes into account not having a default active user set? It looks like it's still going through all of the expenses when it could just exit the functions in case the active user is `''` without doing any calculations.
But will all the boxes still render? imo `Total you paid for` and `Your total share` should not show if you do not have an active user set.

**Comment by meta-boy** [comment]
> I'm also wondering if this takes into account not having a default active user set? It looks like it's still going through all of the expenses when it could just exit the functions in case the active user is `''` without doing any calculations (this is an improvement I would suggest in either case).
> But will all the boxes still render? imo `Total you paid for` and `Your total share` should not show if you do not have an active user set.

Agreed, will make the change

File: src/app/groups/[groupId]/stats/page.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  const expenses = await getGroupExpenses(groupId)
+  const totalGroupSpendings = getTotalGroupSpending(expenses)
+
+  return (
+    <>
+      <Card className="mb-4">
+        <CardHeader>
+          <CardTitle>Totals</CardTitle>
+          <CardDescription>
+          Spending summary of the entire group
```
Comment (scastiel) [inline src/app/groups/[groupId]/stats/page.tsx:39]:
```suggestion
          Spending summary of the entire group.
```

**Review by scastiel — APPROVED** [review]
That’s great, thanks for the contribution! We’ll be able to use the same tab to offer stats by category 😉

**Pull request merged**